screens/LoginScreen.js

```diff
@@ -258,7 +341,29 @@ class LoginScreen extends React.Component {
       );
     }
 
-    if (listsRetrieved === 0) {
+    // contactSettings retrieved
+    if (contactSettings && prevProps.contactSettings !== contactSettings) {
+      AsyncStorage.setItem(
+        'contactSettings',
+        JSON.stringify(contactSettings),
+      );
+    }
+
+    // groupSettings retrieved
+    if (groupSettings && prevProps.groupSettings !== groupSettings) {
+      AsyncStorage.setItem(
+        'groupSettings',
+        JSON.stringify(groupSettings),
+      );
+    }
+
+    if (usersContactsListRetrieved
+      && geonamesListRetrieved
+      && peopleGroupsListRetrieved
+      && searchListRetrieved
+      && usersListRetrieved
+      && contactSettingsListRetrieved
+      && groupSettingsListRetrieved) {
```

Comment:
This is still retrieving all of this data synchronously before it navigates into the app (and even adds more data to wait for). We need to start these requests and navigate immediately before we receive a response from each. The storage of the results for each in AsyncStorage should happen in their reducers.

Reply:
Actually, in the login, 10 end-points are called and the app wait for successfull response from all end-points before redirect to contactList. After the modification, the mobile app will wait for 3 end-points response (login, contactSettings and groupSettings). These last 2 end-point are required to show some texts in the lists views (contacts and groups). The other end-points are processed while the app enters to the contacts view.

Reply:
I would guess we should be able to remove the wait for all but the login, but for now I'll mark this as resolved and we can look at optimizing more in the future.